bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,371 @@
+                video_public,
+            ]
+        else:
+            this_all_keys_dict[video_key] = [
+                False,  # NZSL Absent
+                True,  # S3 PRESENT
+                "",  # gloss_idgloss,
+                "",  # gloss_created_at,
+                "",  # gloss_id
+                "",  # video_id,
+                "",  # gloss_public,
+                "",  # video_public,
+            ]
+
+    # Find NZSL keys that are absent from S3 (present handled above)
+    for video_key, [
+        gloss_idgloss,
+        gloss_created_at,
+        gloss_id,
+        video_id,
+        gloss_public,
+        video_public,
+    ] in this_nzsl_raw_keys_dict.items():
+        if video_key not in this_s3_bucket_raw_keys_list:
+            this_all_keys_dict[video_key] = [
+                True,  # NZSL PRESENT
+                False,  # S3 Absent
+                gloss_idgloss,
+                gloss_created_at,
+                gloss_id,
+                video_id,
+                gloss_public,
+                video_public,
+            ]
+
+    return this_all_keys_dict
+
+
+# Cases
+# In S3     In NZSL     Action
+#   Is        Not         Delete S3 Object
+#   Is        Is          Update ACL
+#   Not       Is          Review
+def get_recommended_action(key_in_nzsl, key_in_s3):
+    if key_in_s3:
+        if key_in_nzsl:
+            return "Update ACL"
+        else:
+            return "Delete S3 Object"
+    return "Review"
+
+
+# Get S3 object's ACL
+def get_s3_canned_acl(video_key):
+    result = aws_cli(
+        [
+            "s3api",
+            "get-object-acl",
+            "--output",
+            "text",
+            "--query",
+            "Grants[*].Permission",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ]
+    )
+    acls_grants = result.stdout.strip().split("\t")
+
+    if len(acls_grants) > 1:
+        if acls_grants[0] == "FULL_CONTROL" and acls_grants[1] == "READ":
+            return "public-read"
+    elif acls_grants[0] == "FULL_CONTROL":
+        return "private"
+
+    return "unknown"
+
+
+# Get S3 object's LastModified date/time
+def get_s3_lastmodified(video_key):
+    result = aws_cli(
+        [
+            "s3api",
+            "head-object",
+            "--output",
+            "text",
+            "--query",
+            "LastModified",
+            "--bucket",
+            AWS_S3_BUCKET,
+            "--key",
+            video_key,
+        ]
+    )
+    return result.stdout.strip()
+
+
+def build_csv_header():
+    return CSV_DELIMITER.join(
+        [
+            "Action",
+            "S3 Video key",
+            "Sbank Gloss created at",
+            "S3 Expected Canned ACL",
+            "S3 Actual Canned ACL",
+            "Sbank Gloss ID",
+            "Sbank Video ID",
+            "Sbank Gloss public",
+            "Sbank Video public",
+            "Sbank Gloss",
+            "S3 LastModified",
+        ]
+    )
```

Comment:
issue: this order is still wrong

it should be:

```
Action
S3 key
S3 Created At
S3 LastModified
Expected Canned ACL
Actual Canned ACL
Gloss ID
Video ID
Gloss Public
Video Public
<any other rows you feel like including>
```

Reply:
Done!